tests/tests/test_brainmapper/test_analysis.py

```diff
@@ -80,3 +191,93 @@ def test_get_region_totals(tmp_path, points, structures_with_points):
         produced_df
     ), "Produced DataFrame no longer matches per-region "
     "count from expected DataFrame"
+
+
+def test_summarise_points_by_region(
+    tmp_path,
+    atlas,
+    points_mouse,
+    points_in_raw_data_space,
+    points_in_atlas_space,
+    volumes_path,
+):
+    """
+    Test that summarise_points_by_atlas_region correctly produces both a
+    points_list csv (summarising coordinates / regions of all points) and
+    a summary csv (containing cell counts / volume summaries per region)
+    """
+
+    points_list_path = tmp_path / "points_list.csv"
+    summary_path = tmp_path / "summary.csv"
+
+    summarise_points_by_atlas_region(
+        points_in_raw_data_space,
+        points_in_atlas_space,
+        atlas,
+        volumes_path,
+        points_list_path,
+        summary_path,
+    )
+
+    # read all results, and original volume csv file
+    points_list_df = pd.read_csv(points_list_path)
+    summary_df = pd.read_csv(summary_path)
+    volumes_df = pd.read_csv(volumes_path)
+
+    counts_by_region = {}
+
+    # Test that each point is correctly written into the points list csv file
+    for index, row in points_list_df.iterrows():
+        assert row.structure_name == points_mouse[index].structure
```

Comment:
Just an FYI, but assert actually takes a second positional argument after the check, which is the error message that will be displayed if the check fails:

```python
assert row.structure_name == points_mouse[index].structure, "Helpful error message here"
```

Not saying to go through and add these custom messages to every statement in this PR, just mentioning since they can be useful when you have lots of `assert` statements that look similar.

Reply:
Thanks @willGraham01 ! I'll bear that in mind for future PRs